File: examples/langchain-code-interpreter/verify.mjs
/**
 * Proves the Tenki-facing part of this example: a LangChain tool that executes
 * Python in a live Tenki sandbox. No LLM needed — we call the tool directly and
 * assert its output. (The full agent, which needs a model key, is in agent.mjs.)
 * Token/project from env (CI) or ~/.config/tenki/config.yaml (local `tenki login`).
 */
import { TenkiSandbox } from "@tenkicloud/sandbox";
import { readFileSync } from "node:fs";
import { homedir } from "node:os";
import { makeCodeTool } from "./tenki-tool.mjs";

const cfg = (key) => {
	try {
		const c = readFileSync(`${homedir()}/.config/tenki/config.yaml`, "utf8");
		return (c.match(new RegExp(`^${key}:\\s*(.+)$`, "m"))?.[1] ?? "").trim();
	} catch {
		return "";
	}
};

const authToken = process.env.TENKI_AUTH_TOKEN || process.env.TENKI_API_KEY || cfg("auth_token");
const projectId = process.env.TENKI_PROJECT_ID || cfg("current_project_id");
const workspaceId = process.env.TENKI_WORKSPACE_ID || cfg("current_workspace_id");
if (!authToken) {
	console.error("No token. Set TENKI_AUTH_TOKEN, or run `tenki login`.");
	process.exit(1);
}

const tenki = new TenkiSandbox({ authToken });
let sandbox;
try {
	sandbox = await tenki.createAndWait({ cpuCores: 1, memoryMb: 1024, projectId, workspaceId });
	const runPython = makeCodeTool(sandbox);

	// Drive the LangChain tool exactly as an agent would.
	const out = await runPython.invoke({ code: "print(sum(range(11)))" });
	if (String(out).trim() !== "55") throw new Error(`tool returned ${JSON.stringify(out)}`);

	console.log("✓ langchain-code-interpreter: LangChain tool executed Python in a Tenki sandbox → 55");
} catch (e) {
	console.error("✗ " + (e?.message ?? e));
	process.exitCode = 1;
} finally {
	if (sandbox) {
		try {
			await sandbox[Symbol.asyncDispose]();
		} catch {
			/* self-reaps via idle/lifetime caps */
		}
	}
}

File: examples/llamaindex/verify.py
"""
Proves the Tenki-facing half of this example: the LlamaIndex tool executes Python
in a live Tenki sandbox. No LLM needed — we call the FunctionTool directly and
assert its output. (The full agent, which needs a model key, is in agent.py.)

Needs Python 3.10+ and the deps in requirements.txt. Token/project from env (CI)
or ~/.config/tenki/config.yaml (local `tenki login`).
"""
import os
import sys

from tenki_sandbox import Sandbox

from tenki_tool import make_code_tool


def cfg(key):
    try:
        with open(os.path.expanduser("~/.config/tenki/config.yaml")) as f:
            for line in f:
                if line.startswith(key + ":"):
                    return line.split(":", 1)[1].strip()
    except Exception:
        pass
    return ""


token = os.environ.get("TENKI_AUTH_TOKEN") or os.environ.get("TENKI_API_KEY") or cfg("auth_token")
if not token:
    print("No token. Set TENKI_AUTH_TOKEN, or run `tenki login`.")
    sys.exit(1)

# The Python SDK sends anything without a known prefix as `Authorization: Bearer`,
# which the server rejects for a browser session token. A `tk_` API key works
# as-is; a `tenki login` session token must be sent as a cookie, so prefix it
# `cookie:`. (SDK auth gap — same note as the other Python examples.)
if not token.startswith(("tk_", "ory_st_", "cookie:")):
    token = f"cookie:{token}"

opts = {"auth_token": token, "cpu_cores": 1, "memory_mb": 1024}
# Project/workspace: env first, then config. `tenki login` usually writes both to
# config.yaml; set TENKI_PROJECT_ID / TENKI_WORKSPACE_ID if yours doesn't.
project = os.environ.get("TENKI_PROJECT_ID") or cfg("current_project_id")
workspace = os.environ.get("TENKI_WORKSPACE_ID") or cfg("current_workspace_id")
if project:
    opts["project_id"] = project
if workspace:
    opts["workspace_id"] = workspace

try:
    with Sandbox.create(**opts) as sb:
        run_python = make_code_tool(sb)
        # Drive the LlamaIndex FunctionTool exactly as an agent would.
        out = run_python.call(code="print(sum(range(11)))")
        if str(out).strip() != "55":
            raise AssertionError(f"tool returned {out!r}")
    print("✓ llamaindex: LlamaIndex FunctionTool executed Python in a Tenki sandbox → 55")
except Exception as e:  # noqa
    print(f"✗ {type(e).__name__}: {e}")
    sys.exit(1)
